File: 338-counting-bits/counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> result;
        for(int i = 0; i <= n ; i++){
            int temp = i;
            int count = 0;
            while(temp > 0){
                if(temp % 2 == 1) count++;
                temp = temp / 2;
            }
            result.push_back(count);
        }
        return result;
    }
};

File: 338-counting-bits/counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        res = []
        for i in range(n+1):
            count = 0
            temp = i
            while temp > 0:
                if temp % 2 == 1:
                    count += 1
                temp = temp // 2
            res.append(count)
        return res
            